packages/edit-site/src/components/style-book/index.js

```diff
@@ -156,65 +245,133 @@ function StyleBook( { isSelected, onSelect, onClose } ) {
+									<EditorStyles styles={ settings.styles } />
+									<style>
+										{
+											// Forming a "block formatting context" to prevent margin collapsing.
+											// @see https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Block_formatting_context
+											`.is-root-container { display: flow-root; }
+											body { position: relative; padding: 32px !important; }` +
+												STYLE_BOOK_IFRAME_STYLES
+										}
+									</style>
+								</>
+							}
+							name="style-book-canvas"
+							tabIndex={ 0 }
+						>
+							{ /* Filters need to be rendered before children to avoid Safari rendering issues. */ }
+							{ settings.svgFilters }
+							<Examples
+								className={ classnames(
+									'edit-site-style-book__examples',
+									{
+										'is-wide': sizes.width > 600,
+									}
+								) }
+								examples={ examples }
+								category={ tab.name }
+								label={ sprintf(
+									// translators: %s: Category of blocks, e.g. Text.
+									__(
+										'Examples of blocks in the %s category'
+									),
+									tab.title
+								) }
+								isSelected={ isSelected }
+								onSelect={ onSelect }
+							/>
+						</Iframe>
 					) }
 				</TabPanel>
 			</section>
 		</StyleBookFill>
 	);
 }
 
-const Examples = memo( ( { examples, category, isSelected, onSelect } ) => (
-	<div className="edit-site-style-book__examples">
-		{ examples
-			.filter( ( example ) => example.category === category )
-			.map( ( example ) => (
-				<Example
-					key={ example.name }
-					title={ example.title }
-					blocks={ example.blocks }
-					isSelected={ isSelected( example.name ) }
-					onClick={ () => {
-						onSelect( example.name );
-					} }
-				/>
-			) ) }
-	</div>
-) );
-
-const Example = memo( ( { title, blocks, isSelected, onClick } ) => (
-	<button
-		className={ classnames( 'edit-site-style-book__example', {
-			'is-selected': isSelected,
-		} ) }
-		aria-label={ sprintf(
-			// translators: %s: Title of a block, e.g. Heading.
-			__( 'Open %s styles in Styles panel' ),
-			title
-		) }
-		onClick={ onClick }
-	>
-		<span className="edit-site-style-book__example-title">{ title }</span>
-		<div className="edit-site-style-book__example-preview">
-			<BlockPreview
-				blocks={ blocks }
-				viewportWidth={ 0 }
-				additionalStyles={ [
-					{
-						css:
-							'.wp-block:first-child { margin-top: 0; }' +
-							'.wp-block:last-child { margin-bottom: 0; }',
-					},
-				] }
-			/>
-		</div>
-	</button>
-) );
+const Examples = memo(
+	( { className, examples, category, label, isSelected, onSelect } ) => {
+		const composite = useCompositeState( { orientation: 'vertical' } );
+		return (
+			<Composite
+				{ ...composite }
+				className={ className }
+				aria-label={ label }
+			>
+				{ examples
+					.filter( ( example ) => example.category === category )
+					.map( ( example, index ) => (
+						<Example
+							key={ example.name }
+							id={ `example-${ index }` }
```

Comment:
Nit: Maybe we should have the `id` unique like a `key` - `example-${ example.name }`? What do you think?

Reply:
Thanks! I was mostly just following the performance example from the reakit docs, but this feels neater and more consistent with how we do things in Gutenberg (using block name as a unique key). Updated in https://github.com/WordPress/gutenberg/pull/48664/commits/f1a2c35466ba8e2f3fad5d67e958878fffca12b4